src/NServiceBus.Core/Transports/Learning/LearningTransportMessagePump.cs

```diff
@@ -88,29 +88,32 @@ public async Task StopReceive(CancellationToken cancellationToken = default)
-
-            await messagePumpTask.ConfigureAwait(false);
-
-            while (concurrencyLimiter.CurrentCount != maxConcurrency)
+            using (cancellationToken.Register(state => (state as CancellationTokenSource)?.Cancel(), messageProcessingCancellationTokenSource))
             {
-                // We are deliberately not forwarding the cancellation token here because
-                // this loop is our way of waiting for all pending messaging operations
-                // to participate in cooperative cancellation or not.
-                // We do not want to rudely abort them because the cancellation token has been canceled.
-                // This allows us to preserve the same behaviour in v8 as in v7 in that,
-                // if CancellationToken.None is passed to this method,
-                // the method will only return when all in flight messages have been processed.
-                // If, on the other hand, a non-default CancellationToken is passed,
-                // all message processing operations have the opportunity to
-                // participate in cooperative cancellation.
-                // If we ever require a method of stopping the endpoint such that
-                // all message processing is canceled immediately,
-                // we can provide that as a separate feature.
-                await Task.Delay(50, CancellationToken.None).ConfigureAwait(false);
+                await messagePumpTask.ConfigureAwait(false);
+
+                while (concurrencyLimiter.CurrentCount != maxConcurrency)
+                {
+                    // We are deliberately not forwarding the cancellation token here because
+                    // this loop is our way of waiting for all pending messaging operations
+                    // to participate in cooperative cancellation or not.
+                    // We do not want to rudely abort them because the cancellation token has been canceled.
+                    // This allows us to preserve the same behaviour in v8 as in v7 in that,
+                    // if CancellationToken.None is passed to this method,
+                    // the method will only return when all in flight messages have been processed.
+                    // If, on the other hand, a non-default CancellationToken is passed,
+                    // all message processing operations have the opportunity to
+                    // participate in cooperative cancellation.
+                    // If we ever require a method of stopping the endpoint such that
+                    // all message processing is canceled immediately,
+                    // we can provide that as a separate feature.
+                    await Task.Delay(50, CancellationToken.None).ConfigureAwait(false);
+                }
             }
 
             concurrencyLimiter.Dispose();
+            messagePumpCancellationTokenSource?.Dispose();
```

Comment:
This is already disposed at the beginning of the method https://github.com/Particular/NServiceBus/pull/6306/files#diff-864733b86acf0f645d84f442925b9047ac9f6fcf72dbdb8549c9a122f3ad94deR87

Reply:
I must have missed that. Will fix

Reply:
I can't see it. I'm probably blind :see_no_evil:

Reply:
Merging. In case it needs more fixes I can submit another PR

Reply:
My bad. It was cancelled at the top, not disposed. I am the one who is blind :)